test/Tests/src/Semantic/Any_Spec.enso

```diff
@@ -42,19 +42,69 @@ spec =
             With_X.Value 44 . x . should_equal "With_X:(With_X.Value 44)"
             With_Y.Value 44 . x . should_equal "With_Y:With_Y(44)"
-            Any.x . to_text . should_equal "Any.type.x[Any_Types.enso:6-26]"
             My_Type.x . should_equal "Any:My_Type"
             With_X.x . to_text . should_equal "Any:With_X"
-            With_X.y . to_text . should_equal "With_X.type.y[Any_Types.enso:12-32]"
+            Meta.should_be_a With_X.y Function
             With_Y.x . to_text . should_equal "Any:With_Y"
-            With_Y.y . to_text . should_equal "With_Y.type.y[Any_Types.enso:18-38]"
+            Meta.should_be_a With_Y.y Function
             With_X.to_text . to_text . should_equal "With_X"
             With_Y.to_text . to_text . should_equal "With_Y"
+            Any.x . to_text . should_equal "Any:Any"
+            Any.x self=Any . to_text . should_equal "Any:Any"
             Any.x self=With_X . should_equal "Any:With_X"
             Any.x self=With_Y . should_equal "Any:With_Y"
             Any.x (My_Type.Value 22) . should_equal "Any:(My_Type.Value 22)"
             Any.x (With_X.Value 22) . should_equal "Any:(With_X.Value 22)"
             Any.x (With_Y.Value 22) . should_equal "Any:With_Y(22)"
             Date.to_display_text . should_equal "Date"
 
+        Test.specify "static method calls on Any should have defaulted self argument to Any" <|
+            # Any.== is a method that takes two arguments (self other)
+            Meta.should_be_a (Any.==) Function
+            Meta.should_be_a (Any.== self=Boolean) Function
+            (Any.== self=Any Boolean) . should_be_false
+            (Any.== Any Any) . should_be_true
+            (Any.== Boolean) . should_be_false
+            (Any.== Any) . should_be_true
+            (Any == Boolean) . should_be_false
+            (Any == Any) . should_be_true
+
+            # Any.to_text is a method that takes one argument (self)
+            Any.to_text . should_equal "Any"
+            Any.to_text self=Any . should_equal "Any"
+            Any.to_text self=Vector . should_equal "Vector"
+            Any.to_text self=[1,2,3] . should_equal "[1, 2, 3]"
+            Any.to_text Any . should_equal "Any"
+            Any.to_text Vector . should_equal "Vector"
+            Any.to_text [1,2,3] . should_equal "[1, 2, 3]"
+
+            Vector.to_text . should_equal "Vector"
+            Text.to_text . should_equal "Text"
+            Number.to_text . should_equal "Number"
+            Integer.to_text . should_equal "Integer"
+
+            # Any.spec_method is a method that takes three arguments (self, arg1, arg2)
+            Meta.should_be_a (Any.spec_method) Function
+            Meta.should_be_a (Any.spec_method Boolean) Function
+            Meta.should_be_a (Any.spec_method self=Boolean) Function
+            Meta.should_be_a (Any.spec_method self=Boolean Vector) Function
+            Any.spec_method Boolean Vector . should_equal "Any.spec_method:{Any}{Boolean}{Vector}"
```

Comment:
This is correct. `self` is defaulted and `Boolean` and `Vector` are the two arguments.

Reply:
Exactly. This is the most important difference from "normal" static method invocation.